stack/stack.py

```diff
@@ -1,25 +1,209 @@
 """
 A stack is a data structure whose primary purpose is to store and
-return elements in Last In First Out order. 
+return elements in Last In First Out order.
 
 1. Implement the Stack class using an array as the underlying storage structure.
    Make sure the Stack tests pass.
 2. Re-implement the Stack class, this time using the linked list implementation
    as the underlying storage structure.
    Make sure the Stack tests pass.
-3. What is the difference between using an array vs. a linked list when 
+3. What is the difference between using an array vs. a linked list when
    implementing a Stack?
 """
+
+# https://www.geeksforgeeks.org/linked-list-vs-array/
+# LL
+  # Memory storage for LL is more dynamic than arrays.
+  # Memory storage for LL is more dynamic than arrays.
+  # insertion and deletion are faster
+  # require more net memory for functionality
+  # memory utilization efficient
+
+# Array
+  # arrays are quicker as have set size
+  # data is contiguous
+  # elements belong to an index
+  # insertion and deletion are more intensive
+  # memory utilization inefficient
+
+
+# class Stack:
+#     def __init__(self):
+#         self.size = 0
```

Comment:
We should increment and decrement on push and pop, respectively, and hold that counter in this variable. Accessing len() of self.storage multiple times could be an issue as the size of the storage gets larger and larger